src/librustc_driver/driver.rs

```diff
@@ -666,13 +675,15 @@ pub fn phase_1_parse_input<'a>(
         profile::begin(sess);
     }
 
+    sess.profiler(|p| p.start_activity(ProfileCategory::Parsing));
```

Comment:
Could you make the `time` function take an `Option<ProfileCategory>` and have it do these calls instead where appropriate?

Reply:
Yes, that would probably work. Good idea!